src/ffi/macros.rs

```diff
@@ -17,7 +17,43 @@ macro_rules! ffi_fn {
         }
     };
 
+    ($(#[$doc:meta])* fn $name:ident($($arg:ident: $arg_ty:ty),*) -> $ret:ty $body:block ?= $default:expr) => {
+        $(#[$doc])*
+        #[no_mangle]
+        pub extern fn $name($($arg: $arg_ty),*) -> $ret {
+            use std::panic::{self, AssertUnwindSafe};
+
+            match panic::catch_unwind(AssertUnwindSafe(move || $body)) {
+                Ok(v) => v,
+                Err(_) => {
+                    eprintln!("panic unwind caught, returning default");
+                    $default
+                }
+            }
+        }
+    };
+
+    ($(#[$doc:meta])* fn $name:ident($($arg:ident: $arg_ty:ty),*) -> $ret:ty $body:block ?: $on_err:expr) => {
```

Comment:
I think, now seeing the `?=` option that has us just return a static expression, that I might prefer that as much as possible, versus passing another function that might panic too... Do you have an example use case in mind? If not, maybe we just stick the `?=` to start with?

Reply:
No. Let's stick to `?=`.

Reply:
Though technically `?=` allows for function calls (that might panic), as long as they return the right type. I don't know how to fix that, as we're using `expr` for default values.

Reply:
Oh yea, it could allow that... Oh well. Still, I don't yet know if we'd want a function to inspect the panic and return a value based on it... Well, maybe to check for OOM, but that can't be done in stable yet anyways.